SpecialContentStaging.php

```diff
@@ -67,30 +89,48 @@ function execute( $par ) {
 						if ( $stage !== "production" ) $target = $keys[$element + 1];
 
 						$currPage = $stages[$stage];
-						$pageNextStage = null;
-						if ( !empty( $target ) ) {
-							$pageNextStage = $stages[$target];
-						}
+						$pageNextStage = $stages[$target];
+
 						$stagingStatus = "<span style=\"color: green\">&#10003;</span>";
 
-						if ( $stage !== "production" && ( !$pageNextStage || $this->replaceStageInternalRefs( $this->pagePrefix, $currPage->getText(), $source, $target ) !== $pageNextStage->getText() ) ) {
-							$stagingStatus = "<html><a href=\"?title=Special:ContentStaging&prefix=" . $this->pagePrefix .
-									"&action=copy&page=" . $currPage->getId() .
-									"&source=" . $source . "&target=" . $target .
-									"\" style=\"color: red;\">&#10007;</a></html>";
+						if( get_class( $currPage ) !== 'WikiPage' ) {
+							$stagingStatus = "";
+						} elseif ( $stage !== "production" && ( get_class( $pageNextStage ) !== 'WikiPage' || $this->replaceStageInternalRefs( $this->pagePrefix, $currPage->getText(), $source, $target ) !== $pageNextStage->getText() ) ) {
```

Comment:
This condition is nice. I think after looking at some code above and below I have figured out what is actually being checked here. How about moving it to a method with some smart name to make it clear?